zallet/src/components/json_rpc/methods/view_transaction.rs

```diff
@@ -450,27 +489,39 @@ pub(crate) async fn call(
     }
 
     if let Some(bundle) = tx.sapling_bundle() {
-        let incoming: BTreeMap<u16, (sapling::Note, Option<sapling::PaymentAddress>, [u8; 512])> =
-            bundle
-                .shielded_outputs()
-                .iter()
-                .zip(0..)
-                .filter_map(|(output, idx)| {
-                    sapling_ivks.iter().find_map(|(ivk, scope)| {
-                        sapling::note_encryption::try_sapling_note_decryption(
-                            ivk,
-                            output,
-                            zip212_enforcement,
-                        )
-                        .map(|(n, a, m)| {
+        let incoming: BTreeMap<
+            u16,
+            (
+                sapling::Note,
+                AccountUuid,
+                Option<sapling::PaymentAddress>,
+                [u8; 512],
+            ),
+        > = bundle
+            .shielded_outputs()
+            .iter()
+            .zip(0..)
+            .filter_map(|(output, idx)| {
+                sapling_ivks.iter().find_map(|(account_id, ivk, scope)| {
+                    sapling::note_encryption::try_sapling_note_decryption(
+                        ivk,
+                        output,
+                        zip212_enforcement,
+                    )
+                    .map(|(n, a, m)| {
```

Comment:
This is another example; we should already have decrypted all of these outputs, and should be able to just get the data from the database - there's no reason we should be trial-decrypting here again.

Reply:
As I've said elsewhere, there *is* a reason: this method provides all information that the wallet can currently decrypt, so if a new key has just been imported but a rescan hasn't proceeded, then this API will still show the relevant spends or outputs in `spends` and `outputs`. After #228 the only thing that won't be accurate until a rescan completes is the `accounts` field, but that doesn't change the fact that it is useful to be able to see the details inside transactions as soon as we have the key material to do so.